lib/src/widgets/flutter_deck_code_highlight.dart

```diff
@@ -0,0 +1,46 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_highlight/flutter_highlight.dart';
+import 'package:flutter_highlight/themes/default.dart';
+import 'package:flutter_highlight/themes/vs2015.dart';
+
+/// This widget provides syntax highlighting for many languages.
+class FlutterDeckCodeHighlight extends StatelessWidget {
+  /// Creates a syntax highlighting widget for displaying code.
+  const FlutterDeckCodeHighlight({
+    required String code,
+    super.key,
+    String? fileName,
+    String language = 'dart',
+  })  : _code = code,
+        _fileName = fileName,
+        _language = language;
+  final String _code;
+  final String? _fileName;
+  final String _language;
+
+  @override
+  Widget build(BuildContext context) => DecoratedBox(
+        decoration: BoxDecoration(
+          color: Theme.of(context).colorScheme.background,
+          borderRadius: const BorderRadius.all(Radius.circular(16)),
+        ),
+        child: Padding(
+          padding: const EdgeInsets.all(8),
+          child: Column(
+            crossAxisAlignment: CrossAxisAlignment.start,
+            mainAxisSize: MainAxisSize.min,
+            children: [
+              if (_fileName != null) Text(_fileName!),
```

Comment:
I would also love to have some spacing here when the `_fileName` is not null (like, SizedBox(height: 4), or something).